src/htm/utils/Topology.cpp

```diff
@@ -227,22 +227,39 @@ bool WrappingNeighborhood::Iterator::operator!=(const Iterator &other) const {
 
 UInt WrappingNeighborhood::Iterator::operator*() const {
   UInt index = 0;
-  for (size_t i = 0; i < neighborhood_.dimensions_.size(); i++) {
+  for (size_t i = 0; i < neighborhood_.ndimensions_; i++) {
     Int coordinate = neighborhood_.centerPosition_[i] + offset_[i];
+    const UInt a = neighborhood_.dimensions_[i]; // the compiler doesn't seem to voluntarily hold on to this value
+
 
     // With a large radius, it may have wrapped around multiple times, so use
     // `while`, not `if`.
 
-    while (coordinate < 0) {
-      coordinate += neighborhood_.dimensions_[i];
-    }
 
-    while (coordinate >= (Int)neighborhood_.dimensions_[i]) {
-      coordinate -= neighborhood_.dimensions_[i];
-    }
+//    while (coordinate < 0) {
+//      coordinate += neighborhood_.dimensions_[i];
+//    }
 
-    index *= neighborhood_.dimensions_[i];
+
+//    while (coordinate >= (Int)neighborhood_.dimensions_[i]) {
+
+
+ //     coordinate -= neighborhood_.dimensions_[i];
+
+
+ //   }
+
+
+    coordinate = coordinate % (Int) a; 
+// C++, depending on the version of the STD, may produce negative "mod n" values.
```

Comment:
oh, good catch! We've spent a lot of time making the code cross-platform and ideally deterministic.